### PR TITLE
DOC: Update scalar types in `Py{TYPE}ArrType_Type`

### PR DESCRIPTION
**Pull request by ghost**
Some scalar types are missing in the list (https://numpy.org/devdocs/reference/c-api/types-and-structures.html#scalararraytypes ).
ref: https://github.com/numpy/numpy/blob/main/numpy/core/code_generators/numpy_api.py#L56-L91

### CODE REVIEW

**Comment by Aryan-Mishra24** [comment]
Assign this to me i'll get it done in an hour

**Comment by charris** [comment]
Not sure about this, I think the list is supposed to be limited to C types.

**Comment by rossbar** [comment]
FWIW the added names do have corresponding C-names - but whether they are intended to be included on this list I don't know.

**Comment by seberg** [comment]
I agree with Chuck.  I suspect this should only list the concrete ones and not the abstract ones here.  That means adding times (and half) would be fine, though.

**Comment by mdhaber** [comment]
@seberg This looks like very low hanging fruit. I take it from your comment that `**TimeInteger**, **Datetime**, **Timedelta**, and **Half**.` can be added to the existing list - any others? (I'm not sure how to distinguish between the concrete and abstract ones.)

**Comment by seberg** [comment]
Abstract are those without instances of course, like `Integer`.

File: doc/source/reference/c-api/types-and-structures.rst [hunk 1]
```diff
@@ -1257,11 +1257,13 @@ that can be present in the array Most of these are simple wrappers
...
-    **Half**, **Float**, **Double**, **LongDouble**, **CFloat**,
-    **CDouble**, **CLongDouble**, **String**, **Unicode**, **Void**, and
-    **Object**.
+    **Bool**, **Generic**, **Number**, **Integer**, **SignedInteger**,
+    **UnsignedInteger**, **Inexact**, **Floating**, **ComplexFloating**,
+    **Flexible**, **Character**, **Byte**, **Short**, **Int**, **Long**,
+    **LongLong**, **UByte**, **UShort**, **UInt**, **ULong**, **ULongLong**,
+    **Float**, **Double**, **LongDouble**, **CFloat**, **CDouble**,
+    **CLongDouble**, **Object**, **String**, **Unicode**, **Void**,
+    **TimeInteger**, **Datetime**, **Timedelta**, and **Half**.
```
Comment (mdhaber) [inline doc/source/reference/c-api/types-and-structures.rst:1266]:
```suggestion
    **Bool**, **Byte**, **Short**, **Int**, **Long**, **LongLong**,
    **UByte**, **UShort**, **UInt**, **ULong**, **ULongLong**,
    **Half**, **Float**, **Double**, **LongDouble**, **CFloat**,
    **CDouble**, **CLongDouble**, **String**, **Unicode**, **Void**, 
    **Datetime**, **Timedelta**, and **Object**.
```

**Comment by mdhaber** [comment]
I'll go ahead and merge this with the addition of `Datetime` and `Timedelta`. If `Timeinteger` or any of the others should also have been added, that can be done separately.

**Pull request merged**